slick/src/main/scala/slick/dbio/DBIOAction.scala

```diff
@@ -63,6 +63,14 @@ sealed trait DBIOAction[+R, +S <: NoStream, -E <: Effect] extends Dumpable {
       (r(0).asInstanceOf[R], r(1).asInstanceOf[R2])
     } (DBIO.sameThreadExecutionContext)
 
+  /** Run another action after this action, if it completed successfully, and zip the result
+    * of both actions with a function `f`, then create a new DBIOAction holding this result,
+    * If either of the two actions fails, the resulting action also fails. */
+  def zipWith[R2, E2 <: Effect,R3](a: DBIOAction[R2, NoStream, E2])(f:(R,R2) =>R3): DBIOAction[R3, NoStream, E with E2] =
+    SequenceAction[Any, ArrayBuffer[Any], E with E2](Vector(this, a)).map { r =>
+      f(r(0).asInstanceOf[R], r(1).asInstanceOf[R2])
```

Comment:
Don't run f on sameThreadExecutionContext. zipWith should take an implicit ExecutionContext, like all other methods that take a user-supplied function.